src/Microsoft.Language.Xml/Syntax/XmlAttributeSyntax.cs

```diff
@@ -111,6 +111,74 @@ public string Value
             }
         }
 
+        /// <summary>
+        /// Get Normalized <see cref="Value"/> of <see cref="XmlStringSyntax"/>
+        /// </summary>
+        /// <remarks>
+        /// Normalization specs:
+        /// <seealso href="https://www.w3.org/TR/2006/REC-xml11-20060816/#sec-line-ends">2.2.12 [XML] Section 3.3.3</seealso/>
+        /// <seealso href="https://learn.microsoft.com/en-us/openspecs/ie_standards/ms-xml/389b8ef1-e19e-40ac-80de-eec2cd0c58ae">2.11 [XML} End-of-Line Handling</seealso/>
+        /// </remarks>
+        public string NormalizedValue
+        {
+            get
+            {
+                if (ValueNode is { } xmlString)
+                {
+                    if (xmlString.TextTokens.Node is null)
+                    {
+                        return "";
+                    }
+
+                    var sb = PooledStringBuilder.GetInstance();

```

Comment:
Let's extract a public static side-effect free helper method `string NormalizeAttributeValue(string)` and move all the normalization logic there. Then call that method from the Value getter.

It's better to extract the side-effect free logic into a separate place, maybe also add a unit-test with some common scenarios.